src/visualization/clustering/k_means_vis.py

```diff
@@ -111,45 +133,76 @@ def __init__(self, data: pd.DataFrame, algorithms_steps: List[Tuple[np.ndarray,
         self.ox_box.addItems(columns)
         self.oy_box = QComboBox()
         self.oy_box.addItems(columns)
+        self.ox_box.currentTextChanged.connect(partial(self.click_listener, 'set_axis'))
+        self.oy_box.currentTextChanged.connect(partial(self.click_listener, 'set_axis'))
         self.settings_box_layout.addRow(QLabel("OX:"), self.ox_box)
         self.settings_box_layout.addRow(QLabel("OY:"), self.oy_box)
 
-        self.axis_button = QPushButton("Submit axis")
-        self.axis_button.clicked.connect(partial(self.click_listener, 'set_axis'))
-        self.settings_box_layout.addRow(self.axis_button)
-
         # visualization layout
         self.visualization_box = QGroupBox()
         self.visualization_box.setTitle("Visualization:")
         self.visualization_box_layout = QVBoxLayout(self.visualization_box)
 
+        self.left_column_layout.addWidget(self.settings_box)
+        self.left_column_layout.addStretch(1)
+
+        if self.animation:
+            # animation
+            self.animation_box = QGroupBox()
+            self.animation_box.setTitle("Animation:")
+            self.animation_box.setFixedWidth(200)
+            self.animation_box_layout = QFormLayout(self.animation_box)
+
+            self.restart_button = QPushButton("Restart")
+            self.restart_button.clicked.connect(partial(self.click_listener, 'restart'))
+            self.run_button = QPushButton("Start animation")
+            self.run_button.clicked.connect(partial(self.click_listener, 'run'))
+            self.interval_box = QSpinBox()
+            self.interval_box.setMinimum(1)
+            self.interval_box.setValue(50)
+            self.interval_box.setMaximum(200)
+
+            self.animation_box_layout.addRow(QLabel("Interval time:"), self.interval_box)
```

Comment:
```suggestion
            self.animation_box_layout.addRow(QLabel("Interval time (ms):"), self.interval_box)
```

I assume that it's in milliseconds